src/input.rs

```diff
@@ -645,163 +562,116 @@ impl State {
             }
             Action::ConsumeWindowIntoColumn => {
                 self.niri.layout.consume_into_column();
-                if self.niri.config.borrow().input.warp_mouse_to_focus {
-                    self.move_cursor_to_focused_tile(CenterCoords::Seperately);
-                }
                 // FIXME: granular
                 self.niri.queue_redraw_all();
             }
             Action::ExpelWindowFromColumn => {
                 self.niri.layout.expel_from_column();
-                if self.niri.config.borrow().input.warp_mouse_to_focus {
-                    self.move_cursor_to_focused_tile(CenterCoords::Seperately);
-                }
                 // FIXME: granular
                 self.niri.queue_redraw_all();
             }
             Action::SwitchPresetColumnWidth => {
                 self.niri.layout.toggle_width();
-                //FIXME: move_cursor_to_focused_tile doesn't work here (reacts to old size)
+                if self.niri.config.borrow().input.warp_mouse_to_focus {
+                    self.move_cursor_to_focused_tile();
```

Comment:
Is it needed on these actions? The focus doesn't change here after all. (And either way, resizes will wait for the client to send a new buffer.)

Reply:
I guess it depends on your workflow whether this is desirable or not. Most of the time nothing will happen anyway. However when resizing a window to a smaller size it can happen that your cursor ends up outside the focused window. Personally I find it convenient if in those cases the cursor warps back inside the focused window. From my testing resize actions don't cause a keyboard or pointer focus change (even in the cases where the cursor ends up outside the focused window), that's why I explicitly added the calls to `move_cursor_to_focused_tile()` here.

Reply:
> From my testing resize actions don't cause a keyboard or pointer focus change (even in the cases where the cursor ends up outside the focused window)

Hm, when the window does submit a new buffer (which is not immediately after a resize action) and the pointer ends up somewhere else, the pointer focus should change.

> that's why I explicitly added the calls to move_cursor_to_focused_tile() here.

Similarly to above, I'm not sure this will do anything because the window will only submit a new, resized buffer sometime later (if it does at all).

Reply:
Tracking things like "niri did a resize action => the window resized *because of that* (and not because it decided to resize on its own)" is not trivial but planned to some extent (will be needed for animations). Then maybe you would be able to put the "cursor ended up outside" check on top of that. For now might be better to do nothing here?

Reply:
> Similarly to above, I'm not sure this will do anything because the window will only submit a new, resized buffer sometime later (if it does at all).

Weirdly enough it does fix it (at least on my machine), but I agree it's a bit of a hack and should probably be left out for now.

Reply:
removed in commit #[94977a0](https://github.com/YaLTeR/niri/pull/245/commits/94977a016963deb9aa2f3dcce15a525357808411)